python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -309,6 +322,25 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
             cdf = cudf.read_parquet(self.paths, columns=with_columns)
             assert isinstance(cdf, cudf.DataFrame)
             df = DataFrame.from_cudf(cdf)
+        elif self.typ == "ndjson":
+            json_schema: list[tuple[str, str, list]] = [
+                (name, typ, []) for name, typ in self.schema.items()
+            ]
+            plc_tbl_w_meta = plc.io.json.read_json(
+                plc.io.SourceInfo(self.paths),
+                lines=True,
+                dtypes=json_schema,
+                prune_columns=True,
+            )
+            # TODO: I don't think cudf-polars supports nested types in general right now
+            # (but when it does, we should pass child column names from nested columns in)
+            df = DataFrame.from_table(
+                plc_tbl_w_meta.tbl, plc_tbl_w_meta.column_names(include_children=False)
+            )
+            # TODO: libcudf doesn't support column-projection (like usecols)
+            # We should change the prune_columns param to usecols there to support this
+            if with_columns is not None:
+                df = df.select(with_columns)
```

Comment:
Check, I think `schema` only contains the projected columns.

Reply:
Yep, double checked and I think schema only contains projected columns and the row-index.